cmd/crc/cmd/daemon_windows.go

```diff
@@ -6,38 +6,42 @@ import (
 
 	"github.com/code-ready/crc/pkg/crc/api"
 	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/pkg/errors"
 	"golang.org/x/sys/windows/svc"
 	"golang.org/x/sys/windows/svc/debug"
 	"golang.org/x/sys/windows/svc/eventlog"
 )
 
-func runDaemon() {
+func runDaemon() error {
 	// Remove if an old socket is present
-	os.Remove(constants.DaemonSocketPath)
-	RunDaemonService("CodeReady Containers", false)
+	_ = os.Remove(constants.DaemonSocketPath)
+	return RunDaemonService("CodeReady Containers", false)
 }
 
 type daemonService struct {
 	eventLogger debug.Log
 }
 
 func (m *daemonService) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
-	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
-	crcAPIServer, err := api.CreateServer(constants.DaemonSocketPath, newConfig, newMachineWithConfig)
+	apiServer, err := api.CreateServer(constants.DaemonSocketPath, newConfig, newMachineWithConfig)
 	if err != nil {
-		_ = m.eventLogger.Error(1, fmt.Sprintf("Failed to start CRC daemon service: %v", err))
+		_ = m.eventLogger.Error(1, fmt.Sprintf("Failed to start daemon service: %v", err))
 		return
 	}
-	// Windows servivce manager calls OnStart when starting the service
+	// Windows service manager calls OnStart when starting the service
 	// and it calls this method, which the service manager expects to handle
-	// commands like stop, shutdown. crcAPIServer.Serve is being called as a go routine
+	// commands like stop, shutdown. apiServer.Serve is being called as a go routine
 	// since we don't want it block
-	go crcAPIServer.Serve()
-	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+	go func() {
+		if err := apiServer.Serve(); err != nil {
+			_ = m.eventLogger.Error(1, fmt.Sprintf("Failed to run daemon: %v", err))
+		}
+	}()
+	changes <- svc.Status{State: svc.Running, Accepts: svc.AcceptStop | svc.AcceptShutdown}
 
 	m.loop(r, changes)
 
-	// Above MainLoop will return when the daemon service is stopped
+	// Above loop will return when the daemon service is stopped
 	changes <- svc.Status{State: svc.Stopped}
```

Comment:
lots of changes in this file which seem related to the removal of 'crc' in type names rather being related to returning an error from `runDaemon`.